roles/nginx/templates/alfresco.conf.j2

```diff
@@ -0,0 +1,45 @@
+map $remote_addr $solr_trackers {
+    default     0;
+{% for solr_tracker in [ solr_host ] | unique %}
+    {{ solr_tracker }}	   1;
+{% endfor %}
+}
+
+upstream repo_lb {
+        ip_hash;
+{% for repo in repo_hosts | unique %}
+{% if not repo.cluster_keepoff | default(false) %}
+        server {{ repo.local_addr | default(repo.inventory_name) }}:{{ ports_cfg.repository.http | default(8080) }} max_fails=2 fail_timeout=8s;
+{% endif %}
+{% endfor %}
+    }
+
+upstream tracker_lb {
+{% for repo in repo_hosts | unique %}
+        server {{ repo.local_addr | default(repo.inventory_name) }}:{{ ports_cfg.repository.http | default(8080) }} max_fails=2 fail_timeout=8s {% if not repo.cluster_keepoff | default(false) and repo_hosts | selectattr('cluster_keepoff', 'true') %}backup{% endif %}; 
```

Comment:
I am not getting entirely the if here for handling backup flag:

if there are 3 nodes and one of them is marked as keepoff, this one get declared as backup

while the first condition of the if is for ensuring what?

Reply:
still not convinced that `not repo.cluster_keepoff | default(false)` is what you had explained, or maybe I've troubles regarding precedence of NOT operator in jijna, I try with an example:

if I have:
repo1.cluster_keepoff = false
repo2.cluster_keepoff = false
repo3.cluster_keepoff = true

I obtain:
repo1 backup
repo2 backup
repo3

what I am not getting?

Reply:
you're getting it right. That's indeed how the template renders (and needs to). That LB is not a LB it's just a failover which prefers talking to the node that's not part of the cluster if any is present

Reply:
ok, the last phrase makes me understand what we are trying to achieve, I was thinking that we need to do the opposite  (flag backup only the keepoff cluster)